pallets/proofs-dealer/src/lib.rs

```diff
@@ -271,6 +283,19 @@ pub mod pallet {
     #[pallet::getter(fn slashable_providers)]
     pub type SlashableProviders<T: Config> = StorageMap<_, Blake2_128Concat, ProviderIdFor<T>, ()>;
 
+    /// A mapping from block number (tick) to Providers, which is set if the Provider submitted a valid proof in that tick.
+    ///
+    /// This is used to keep track of the Providers that have submitted proofs in the last few
+    /// ticks, where availability only up to the last `TargetTicksStorageOfSubmitters` ticks is guaranteed.
+    /// This storage is then made available for other pallets to use through the `ReadProofSubmittersInterface`.
+    #[pallet::storage]
+    pub type ValidProofSubmittersLastTicks<T: Config> = CountedStorageMap<
```

Comment:
Is it really worth it using a `CountedStorageMap`? It adds complexity to benchmarking reads and writes.

Reply:
If we add the variable that keeps track of the last deleted tick we don't need to know how many ticks are stored, we can just make sure we only delete up to the `ChallengesTicker - TargetTicksStorageOfSubmitters` tick